ethicml/data/tabular_data/adult.py

```diff
@@ -149,91 +150,38 @@ def adult(
+        class_label_spec = "salary_>50K"
         class_label_prefix = ["salary"]
     elif split == "Race-Binary":
-        sens_attrs = ["race_White"]
+        sens_attr_spec = "race_White"
         s_prefix = ["race"]
-        class_labels = ["salary_>50K"]
+        class_label_spec = "salary_>50K"
         class_label_prefix = ["salary"]
     elif split == "Custom":
-        sens_attrs = []
+        sens_attr_spec = {}
         s_prefix = []
-        class_labels = []
+        class_label_spec = {}
         class_label_prefix = []
     elif split == "Race-Sex":
-        # TODO: this is currently broken. we'd need to take into account that race is one-hot
-        sens_attrs = [
-            "sex_Male",
-            "race_Amer-Indian-Eskimo",
-            "race_Asian-Pac-Islander",
-            "race_Black",
-            "race_Other",
-            "race_White",
-        ]
+        sens_attr_spec = simple_spec({"sex": ["sex_Male"], "race": disc_feature_groups["race"]})
         s_prefix = ["race", "sex"]
-        class_labels = ["salary_>50K"]
+        class_label_spec = "salary_>50K"
         class_label_prefix = ["salary"]
     elif split == "Nationality":
-        sens_attrs = [
-            "native-country_Cambodia",
-            "native-country_Canada",
-            "native-country_China",
-            "native-country_Columbia",
-            "native-country_Cuba",
-            "native-country_Dominican-Republic",
-            "native-country_Ecuador",
-            "native-country_El-Salvador",
-            "native-country_England",
-            "native-country_France",
-            "native-country_Germany",
-            "native-country_Greece",
-            "native-country_Guatemala",
-            "native-country_Haiti",
-            "native-country_Holand-Netherlands",
-            "native-country_Honduras",
-            "native-country_Hong",
-            "native-country_Hungary",
-            "native-country_India",
-            "native-country_Iran",
-            "native-country_Ireland",
-            "native-country_Italy",
-            "native-country_Jamaica",
-            "native-country_Japan",
-            "native-country_Laos",
-            "native-country_Mexico",
-            "native-country_Nicaragua",
-            "native-country_Outlying-US(Guam-USVI-etc)",
-            "native-country_Peru",
-            "native-country_Philippines",
-            "native-country_Poland",
-            "native-country_Portugal",
-            "native-country_Puerto-Rico",
-            "native-country_Scotland",
-            "native-country_South",
-            "native-country_Taiwan",
-            "native-country_Thailand",
-            "native-country_Trinadad&Tobago",
-            "native-country_United-States",
-            "native-country_Vietnam",
-            "native-country_Yugoslavia",
-        ]
+        sens = "native-country"
+        sens_attr_spec = simple_spec({sens: disc_feature_groups[sens]})
```

Comment:
this is the code I was talking about

```suggestion
        sens_attr_spec = simple_spec({(key := "native-country"): disc_feature_groups[key]})
```